src/api/procedures/setTransferRestrictions.ts

```diff
@@ -18,88 +25,128 @@ import {
   TxTag,
   TxTags,
 } from '~/types';
-import { ProcedureAuthorization, StatisticsOpType } from '~/types/internal';
+import { ProcedureAuthorization } from '~/types/internal';
 import {
   complianceConditionsToBtreeSet,
-  scopeIdsToBtreeSetIdentityId,
-  statisticsOpTypeToStatOpType,
-  stringToIdentityId,
+  identitiesToBtreeSet,
+  identityIdToString,
+  meshClaimTypeToClaimType,
   stringToTickerKey,
   toExemptKey,
   transferRestrictionToPolymeshTransferCondition,
+  transferRestrictionTypeToStatOpType,
   u32ToBigNumber,
 } from '~/utils/conversion';
 import {
   assertStatIsSet,
   checkTxType,
   compareTransferRestrictionToInput,
-  getExemptedIds,
   neededStatTypeForRestrictionInput,
 } from '~/utils/internal';
 
 /**
  * @hidden
  */
+
+type ClaimKey = keyof typeof ClaimType | 'None';
+type ExemptionRecords = Partial<Record<ClaimKey, Identity[]>>;
+
+const addExemptionIfNotPresent = (
+  toInsertId: Identity,
+  exemptionRecords: ExemptionRecords,
+  claimType: ClaimKey,
+  filterSet: Identity[] = []
+): void => {
+  const found = filterSet.find(currentId => currentId.did === toInsertId.did);
```

Comment:
```suggestion
  const found = filterSet.some(currentId => currentId.isEqual(toInsertId));
```
I think `some` makes more sense here, since we don't actually need the matching value. Also, we can directly use `isEqual` (can leave for direct did assertion as well)